### PR TITLE
Feature: Added Taskbar Search & Sort to Monthly View

### LINKED ISSUE #72: Create Task Selector for Taskbar with Sorting Functions
## Description
Implement a task selector for the taskbar on the left side of the application. This selector should provide two functions: one to return an array of tasks sorted by due date (earliest first), and another to return an array of completed tasks sorted by importance (most important on a scale of 0-100 where 100 is most important should be first).

## Acceptance Criteria
- [x] Task selector is implemented in the taskbar on the left side.
- [x] Function to return an array of tasks sorted by due date (earliest first) is created.
- [x] Function to return an array of completed tasks sorted by importance (most important first) is created.
- [x] Tasks are displayed correctly in the taskbar according to the selected sorting function.

## Tasks
- [x] Design the task selector interface for the taskbar.
- [x] Implement the function to return an array of tasks sorted by due date (earliest first).
- [x] Implement the function to return an array of completed tasks sorted by importance (most important first).
- [x] Integrate the task selector with the taskbar.
- [x] Ensure tasks are displayed correctly in the taskbar based on the selected sorting function.

## Context
This feature will enhance the usability of the taskbar by allowing users to easily view tasks sorted by due date and importance. It will help users prioritize their tasks more effectively.

## Screenshots/Mockups
_Include any relevant screenshots or mockups if applicable._

## Testing Notes
- Verify that the task selector correctly returns an array of tasks sorted by due date (earliest first).
- Verify that the task selector correctly returns an array of completed tasks sorted by importance (most important first).
- Check the display of tasks in the taskbar to ensure they are sorted correctly according to the selected function.

## Assignees
/assign @Realizedd @Mashyuf @andrewtran5222 

## Labels
/label enhancement, documentation

### PR DESCRIPTION
**Pull request by Realizedd**
Closes #72

### CODE REVIEW

File: source/monthly-planner-taskbar.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        "task-time": "11:59 PM",
+        "task-title": "Checkpoint Video Due",
+        "task-description":
+            "Submit our checkpoint video by the end of the day.",
+        importance: 0.3,
+        color: "#e69996",
+        notes: "Make sure to include all team members in the video.",
+        completed: true,
+    },
+];
```
Comment (lauT0) [inline source/monthly-planner-taskbar.js:24]: good example, but we will need to remove hardcoding gin the future. The task should be read in from the tasks.json file.

File: source/monthly-planner-taskbar.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    if (switchToWeeklyButton) {
+        switchToWeeklyButton.addEventListener("click", () => {
+            window.location.href = escape("planner.html");
+        });
+    }
+}
+
+function refreshTaskList(options) {
+    const taskList = document.querySelector("#taskbar-list");
+    taskList.innerHTML = "";
```
Comment (lauT0) [inline source/monthly-planner-taskbar.js:98]: innerHTML method is outdated.

File: source/monthly-planner.html [hunk 3]
```diff
@@ -5,39 +5,63 @@
...
+                        id="add-button"
+                        class="navigator-button edit-button"
+                    >
+                        +
+                    </button>
+                    <button id="prev-button" class="navigator-button">
                         &#x3008;
                     </button>
-                    <button id="curButton" class="navigator-button">
+                    <button id="cur-button" class="navigator-button">
```
Comment (lauT0) [inline source/monthly-planner.html:61]: good standardizing of these ids, we probably need to make more improvements like this across the whole repo

File: source/monthly-planner.html [hunk 4]
```diff
@@ -50,113 +74,12 @@ <h1 class="month-title">May 2024</h1>
                 <li class="day-name">Thu</li>
                 <li class="day-name">Fri</li>
                 <li class="day-name">Sat</li>
-                <!-- original template
```
Comment (lauT0) [inline source/monthly-planner.html:53]: good to delete unnecessary comments

**Comment by wsugiarto** [comment]
Added:
Taskbar Search & Sort: Functionality to search and sort tasks in the monthly view.
Event Listeners: Handlers for search bar input, sort order selection, completion status, and date range changes.
Improved:
Refactored HTML Structure: Updated the taskbar and search components for better integration and usability.
Code Refactoring: Improved the readability and maintainability of the taskbar-related JavaScript code.
Security Fixes:
XSS Prevention: Addressed cross-site scripting (XSS) vulnerabilities.
Potential Concerns:
InnerHTML Usage: Using innerHTML method, which may pose security risks if not handled carefully.
Hardcoded Tasks: Initial tasks are hardcoded in the JavaScript file, which is not ideal for dynamic applications.

**Pull request merged**